packages/core/src/sink/IndexSink.ts

```diff
@@ -0,0 +1,41 @@
+/** @license MIT License (c) copyright 2010-2016 original author or authors */
+/** @author Brian Cavalier */
+/** @author John Hann */
+
+import Pipe from './Pipe'
+import { Time, Sink } from '@most/types'
+
+export interface IndexedValue<A> {
+  readonly index: number
+  readonly value: A
+  readonly active: boolean
+}
+
+export default class IndexSink<A> extends Pipe<A, Readonly<IndexedValue<A | undefined>>> implements Sink<A> {
+  readonly index: number
```

Comment:
Does `readonly` really provide any value? See https://www.typescriptlang.org/play/#code/MYGwhgzhAEBSCmATAltA3gWAFDV9ATvGIgPYB2IAntGWALbwBc0EALvsmQObbZ7TBybfAFdgrEvgAUtBs2Gcu0ALzQA5AGUA1pQDuYEFvj41ASnR9+eVgAtkEAHSz4KmvXiXcAX2w+s2TlZjADMwYBcAIUIwLQAlIlIKakwcPGd5dkUAbl9eLAA3MHxoACskVFUyeF04cqlTHP8sQTIIEhB4BxASLikylCd3BugAehHtPQMjfGw+8uhIaCiiOITyKlNBhlc1AE0SRDA1Rpa2jq6euYHnYbH9w6A

Reply:
Yeah, it does (as long as you don't cast 😂). It's js anyway, any constraint can be broken.

Reply:
So, leaving as is? `readonly` still helps a lot if you don't cheat :)

Reply:
I guess so ;)